### PR TITLE
Improve the Android release documentation

### PR DESCRIPTION
**Pull request by SandroMachado**
The current documentation for the `Android` platform for the release process is outdated. This PR fixes the documentation.

### CODE REVIEW

**Comment by lacker** [comment]
@grabbou would you mind reviewing this?

File: docs/SignedAPKAndroid.md [hunk 1]
```diff
@@ -78,6 +78,8 @@ Simply run the following in a terminal:
 $ cd android && ./gradlew assembleRelease
 ```
 
+> If you prefer you can just run `react-native run-android --configuration=release` that the **release** `apk` will be generated and installed into the connected `android` device.
```
Comment (grabbou) [inline docs/SignedAPKAndroid.md]: Not sure if this is grammatically correct, `that the **release** `apk``, shouldn't be `that will generate the release apk`?

Comment (grabbou) [inline docs/SignedAPKAndroid.md]: Also, I believe it's installed onto, not into.

**Comment by SandroMachado** [comment]
@grabbou update! Thanks for the review.

**Comment by grabbou** [comment]
BTW where is the `configuration` flag coming from? In the run-android docs I can still see the `variant`.

**Comment by SandroMachado** [comment]
@grabbou [this](https://github.com/facebook/react-native/commit/85c8333bc87d27a7f5f31a74d62c3d0b3ad631f9) commit destroyed the `configuration` flag and hardcoded the `installDebug`, the new feature introduced in [0.39](https://github.com/facebook/react-native/commit/2a2d3c6e5a54c553a5c8a9932d8204463e7bbb49). This is a regression.... Can this commit be removed? Being able to generate release builds from terminal is very useful.

**Comment by shawngustaw** [comment]
Just wanted to chime in that it'd be helpful if this got merged :)

**Comment by jpshelley** [comment]
It seems configuration works great in React Native `0.40.0` (current stable) and variant doesn't as described in the issue, but in `0.41.0-rc.0` variant works as it should again and configuration doesn't.
Should probably just stick with one so as not to confuse future versions.

**Comment by SandroMachado** [comment]
@grabbou sorry to bother you again. Can you please take a look to this PR and to the regressions described here?

**Comment by SandroMachado** [comment]
@hramos can you please take a look at my concerns described here: https://github.com/facebook/react-native/pull/11443#issuecomment-268214342?

**Comment by hramos** [comment]
Sorry I missed that - will revert this to "needs revision" until that feedback is addressed.

**Comment by hramos** [comment]
Sorry I missed that - from what I can see, this does not prevent the PR from being merged. Should we open a new issue to track this?

**Comment by SandroMachado** [comment]
The problem is that the feature I am documenting is on 0.40 and 0.39 but it is not in master (and in 0.41 beta). It was removed (I think, by mistake or in a bad rebase) by this commit: https://github.com/facebook/react-native/commit/85c8333bc87d27a7f5f31a74d62c3d0b3ad631f9

My suggestion is to remove this commit: https://github.com/facebook/react-native/commit/85c8333bc87d27a7f5f31a74d62c3d0b3ad631f9 rebase the PR with that commit and merge it again, but this time, without removing the `configuration` flag.

I think the `configuration` flag should not be removed. There are a lot of developers already using it like I said in my previous comment.

**Comment by SandroMachado** [comment]
@hramos and @grabbou with `0.41.X` release the `--configuration` for android was removed by the bad merge I am talking above.

I can make a PR reintroducing the flag. WDYT?

**Comment by SandroMachado** [comment]
@grabbou and @hramos sorry to bother you again.

I decided to fix the issues I am talking above in another [PR](#12402). 

I will close this PR, since it is deprecated by #12402.

**Pull request closed without merging**